### PR TITLE
Add "one billion row challenge" study

### PR DESCRIPTION
**Pull request by stonea**
This PR includes a couple of Chapel written versions of the "One Billion Row Challenge" (https://github.com/gunnarmorling/1brc).

Currently, in these versions we're not trying to get the absolute best performance possible (by for example mmaping data or writing custom hashing routines as is seen in many performant implementations in other languages).

Rather we plan to use this as an example to study parallel IO performance and programmability. We might come back and look at other performance tricks later.

The two versions are:

**aggregators** - (written by @jeremiah-corrado), which uses a custom accumulation aggregator.
**parSafeMapVer** - uses Chapel's map class configured to be "parallel safe". I also have a `config const` that can optionally run this  in a serial manner.

I also include a very straightforward (non performant) Python version `python_version.py`, which we may want to compare against.

This PR also adds nightly performance testing for the two Chapel versions and a serial chapel version. On my laptop for the million row dataset I see it takes:

* 0.6 s for the serial version
* 0.44 s for the parallel safe map version
* 0.37 s for the aggregators based version

For testing purposes I add a scaled down one million row file (approx. 13 MB). Building the official Java versions might be a bit of a hassle (we'd need to get a specific version of Java installed on the machines we use for nightly testing). Alternatively, we could use this unofficial Python script (https://github.com/gunnarmorling/1brc/blob/main/src/main/python/create_measurements.py), which is Apache licensed though I'm not sure I want to jump through the hoops to get it included.

### CODE REVIEW

File: test/studies/1brc/aggregators.good [hunk 1]
```diff
@@ -0,0 +1,415 @@
+aggregators.chpl:86: In method 'flush':
+aggregators.chpl:88: warning: borrowed sharedLock(map(bytes,tempData,false)) is being used in a 'manage' statement via its 'enterContext' and 'exitContext' methods. However, borrowed sharedLock(map(bytes,tempData,false)) does not implement contextManager. In the future, this will result in an error.
```
Comment (cassella) [inline test/studies/1brc/aggregators.good:2]: This seems a bit concerning.

**Review by jeremiah-corrado — APPROVED** [review]
Looks good to me! I'll open an issue about the erroneous interfaces warning.

My only concern is that the 1M line input file seems pretty large to include in our tree. I think it might be nicer to use a 1000ish line file for correctness tests, and then generate a 1-10M line file on the fly for perf tests. There is a [python script](https://github.com/gunnarmorling/1brc/blob/main/src/main/python/create_measurements.py) for generating the input data so we wouldn't need to worry about having java installed wherever the test is run.

File: test/studies/1brc/aggregators.good (hunk 1 above)
Comment (jeremiah-corrado) [reply test/studies/1brc/aggregators.good:2]: This is a bug related to how that warning is generated. See: https://github.com/chapel-lang/chapel/issues/24408

**Comment by stonea** [comment]
> There is a [python script](https://github.com/gunnarmorling/1brc/blob/main/src/main/python/create_measurements.py) for generating the input data so we wouldn't need to worry about having java installed wherever the test is run

I've created an executable `SKIPIF` to download, patch, and run that script to dynamically generate the dataset.

File: test/studies/1brc/aggregators.perfkeys [hunk 2]
```diff
@@ -0,0 +1 @@
+elapsed time:
```
Comment (jeremiah-corrado) [inline test/studies/1brc/aggregators.perfkeys:1]: I might add a [verify line](https://chapel-lang.org/docs/developer/bestPractices/TestSystem.html#validating-performance-test-output) to this too:
```
verify:-1:     İzmir: -18.6 17.9 54.0
```

**Review by jeremiah-corrado — APPROVED** [review]

File: test/studies/1brc/aggregators.perfkeys (hunk 2 above)
Comment (stonea) [reply test/studies/1brc/aggregators.perfkeys:1]: I think there's some amount of randomness to the input set (now that we're generating it dynamically) so verifying that way wouldn't work. Though we could rig it up to use some fixed seed if we wanted.

**Pull request merged**